test/test_complex.jl

```diff
@@ -323,4 +323,18 @@ function test_hermitian_jump_scalar()
     return
 end
 
+function test_mul_real_hermitian()
+    A = [1 1+1im; 1-1im 2]
+    model = Model()
+    @variable(model, x)
+    for s in (:L, :U), f in (x, x + 1, x^2)
+        B = LinearAlgebra.Hermitian(A, s)
+        @test f * B isa LinearAlgebra.Hermitian
+        @test isequal_canonical(f * B, LinearAlgebra.Hermitian(f * A))
+        @test B * f isa LinearAlgebra.Hermitian
+        @test isequal_canonical(B * f, LinearAlgebra.Hermitian(A * f))
```

Comment:
I'd suggest using an underlying non-Hermitian matrix for a more stringent test. I've seen related bugs appearing in `LinearAlgebra`.
```suggestion
    A = [1+1im 1+1im; 1-2im 3+1im]
    model = Model()
    @variable(model, x)
    for s in (:L, :U), f in (x, x + 1, x^2)
        B = LinearAlgebra.Hermitian(A, s)
        @test f * B isa LinearAlgebra.Hermitian
        @test isequal_canonical(f * B, LinearAlgebra.Hermitian(f * A, s))
        @test B * f isa LinearAlgebra.Hermitian
        @test isequal_canonical(B * f, LinearAlgebra.Hermitian(A * f, s))
```